File: packages/admin/configs/app-settings.js
module.exports = {
  singleton: true,
  fields: [
    {
      label: 'Announcement',
      key: 'announcement',
      group: true,
      fields: [
        {
          label: 'Enable',
          key: 'enable',
          type: 'switch',
          size: 'half',
        },
        {
          size: 'half',
        },
        {
          label: 'Image',
          key: 'cover_image',
          type: 'fileUpload',
          size: 'half',
        },
        {
          label: 'Description',
          key: 'description',
          type: 'textarea',
          size: 'half',
        },
      ]
    },
    {
      label: 'Booking Conditions',
      key: 'booking_conditions',
      type: 'textarea',
    }
  ],
  sidebarFields: [
    {
      label: 'Server status (for application)',
      key: 'server_status',
      type: 'select',
      size: 'half',
      options: [
        {
          label: 'Live',
          value: 'live',
        },
        {
          label: 'Maintenance',
          value: 'maintenance',
        },
      ],
    },
    {
      label: 'Phone',
      key: 'phone',
      type: 'text',
    },
    {
      label: 'Line',
      key: 'line',
      type: 'text',
    },
  ],
  editTitle: 'Application Settings',
  initValue: {
    content: [],
  },
}

File: packages/application/lib/models/app_settings.dart
import 'package:chongchana/models/announcement.dart';

enum ServerStatus {
  live,
  maintenance,
}

class ApplicationSettings {
  late String bookingConditions;
  late String line;
  late String phone;
  late ServerStatus serverStatus;
  late Announcement? announcement;

  ApplicationSettings({
    this.bookingConditions = "",
    this.announcement,
    this.line = "",
    this.phone = "",
    this.serverStatus = ServerStatus.live,
  });
  ApplicationSettings.fromJson(Map<String, dynamic> json) {
    try {
      bookingConditions = json["booking_conditions"];
      serverStatus = json["server_status"] == "maintenance"
          ? ServerStatus.maintenance
          : ServerStatus.live;
      line = json["line"] ?? "";
      phone = json["phone"] ?? "";

      if (serverStatus == ServerStatus.maintenance) {
        announcement = Announcement(
            isMaintenance: true,
            enable: true,
            description: "Server is maintenance");
      } else {
        announcement = Announcement.fromJson(json["announcement"]);
      }
    } catch (e) {
      print("Error from ApplicationSettings class: $e");
    }
  }
}
